### DISCUSSION TITLE
Chain switcher before connecting wallet

### DISCUSSION THREAD

**Discussion by art-eth** [rainbow-me/rainbowkit#1668] [Ideas]
Continuing discussion from PR https://github.com/rainbow-me/rainbowkit/pull/1601
cc @KosmosKey 

What would be benefits if it was implemented on RainbowKit side?
- better support for multi-chain websites out of the box while using RainbowKit
- same user experience for switching chains before/after connecting wallet and a devs would be thankful for not having to create that modal themselves ;)

I think you (@KosmosKey) might have misunderstood a bit my usecase. I requested that feature so users can easily change "default chain" if wallet is not being connected - if it's connected then RainbowKit uses the chain selected by the user in his wallet (MetaMask or whatever they use) .

**Here is a simple usecase example:**
I have a website with farming pools on 2 chains - Ethereum and Polygon. Right now `getPublicClient()` returns the first chain defined in `RainbowKitProvider`'s `chains` property so by default I fetch data from the `Ethereum` chain in this case. To fetch anything from the other supported chain on my website, I need to either ask suer to connect their wallet (so they have chain switcher available) or as you mentioned - create my own chain's switcher which is a lot effort to make it look similar etc.

**Possible solution:**
Allow user's to switch chain before connecting wallet if there is more than 1 supported chain so that way `getPublicClient()` could return the chain selected by user on website instead of defaulting to the first one that was configured in an array. 
If possible, `const { chain } = useNetwork()` should also return the chain selected by user (defaulting to the first one that was configured), right now it's `underfined` until connecting the wallet.

**Comment by art-eth** [comment]
cc @oxSaturn @hzhu

**Comment by oxSaturn** [comment]
Copied from https://github.com/rainbow-me/rainbowkit/discussions/1671

Here's why I think it's better UX to show it without wallet connected.

1. users can know which chains are supported without wallet connected, take uniswap for example,
![CleanShot-JGgdbuy0@2x](https://github.com/rainbow-me/rainbowkit/assets/126733611/c18bd9ec-2392-46a8-86f5-5a41bf151bff)

    As you can see, I can tell it's deployed on which chains without having a wallet installed or connected, while with rainbowkit, it's not possible, it's kind of a paywall there.

3. users can see the current connected chain set by the `initialChain`, currently there's no way to tell which chain is it without connected. Probably you can check the wallets like rabby etc., but that's not accurate as it can be different from the one dapp uses with the `initialChain`.

**Comment by art-eth** [comment]
@KosmosKey wagmi supports that

> When connected, switchChain will switch the target chain for the connector. When not connected, switchChain will switch the target chain for the Wagmi [Config](https://rc.wagmi.sh/core/createConfig#config).

https://rc.wagmi.sh/core/api/actions/switchChain#usage

**Reply by DanielSinclair** [reply]
@0xartcro Thanks for pointing out this change; we'll evaluate this once the Wagmi v2 migration is finished.

**Comment by oxSaturn** [comment]
Related PR https://github.com/rainbow-me/rainbowkit/pull/1709

**Comment by vikaman2103** [comment]
I have to disable other browser wallets so that they don't pop up without a request.
